[Go Binding] EXPORT_SYMBOL overlap with embedder.h

```bash
In file included from wrapper.cpp:13:
wrapper.h:7: warning: "EXPORT_SYMBOL" redefined
    7 | #define EXPORT_SYMBOL __declspec(dllexport)
      | 
In file included from wrapper.cpp:12:
../../src/embedder.h:17: note: this is the location of the previous definition
   17 |         #define EXPORT_SYMBOL __declspec(dllimport)
      | 
wrapper.cpp: In function 'HINSTANCE__* load_library(const char*)':
```

--- a/.github/workflows/go-test.yml
+++ b/.github/workflows/go-test.yml
@@ -86,3 +86,8 @@ jobs:
           SHARED_LIB_PATH: ${{ env.SHARED_LIB_PATH }}
           CMAKE_FLAGS: ${{ env.CMAKE_FLAGS }}
           CMAKE_BUILD_FLAGS: ${{ env.CMAKE_BUILD_FLAGS }}
+      - name: Setup tmate session
+        if: always() && matrix.os == 'windows-latest'
+        uses: mxschmitt/action-tmate@v3
+        with:
+          limit-access-to-actor: true

--- a/bindings/go/wrapper.cpp
+++ b/bindings/go/wrapper.cpp
@@ -41,21 +41,25 @@ std::string GetLastErrorAsString() {
 #endif
 
 #if defined(_WIN32) || defined(_WIN64)
-
-typedef llama_embedder * (__cdecl *init_embedder_local_func)(const char *, uint32_t);
-typedef void (__cdecl *free_embedder_local_func)(llama_embedder *);
-typedef FloatMatrix (__cdecl *embed_c_local_func)(llama_embedder *, const char  ** , size_t , int32_t);
-typedef int (__cdecl *get_metadata_c_local_func)(llama_embedder *,MetadataPair**, size_t*);
-typedef void (__cdecl *free_metadata_c_local_func)(MetadataPair*, size_t);
-
+#if defined(_WIN64)
+        typedef llama_embedder* (*init_embedder_local_func)(const char*, uint32_t);
+        typedef void (*free_embedder_local_func)(llama_embedder*);
+        typedef FloatMatrix (*embed_c_local_func)(llama_embedder*, const char**, size_t, int32_t);
+        typedef int (*get_metadata_c_local_func)(llama_embedder*, MetadataPair**, size_t*);
+        typedef void (*free_metadata_c_local_func)(MetadataPair*, size_t);
+    #else
+        typedef llama_embedder* (__cdecl *init_embedder_local_func)(const char*, uint32_t);
+        typedef void (__cdecl *free_embedder_local_func)(llama_embedder*);
+        typedef FloatMatrix (__cdecl *embed_c_local_func)(llama_embedder*, const char**, size_t, int32_t);
+        typedef int (__cdecl *get_metadata_c_local_func)(llama_embedder*, MetadataPair**, size_t*);
+        typedef void (__cdecl *free_metadata_c_local_func)(MetadataPair*, size_t);
+    #endif
 #else
-
-typedef llama_embedder * (*init_embedder_local_func)(const char *, uint32_t);
-typedef void (*free_embedder_local_func)(llama_embedder *);
-typedef FloatMatrix (*embed_c_local_func)(llama_embedder *, const char  ** , size_t , int32_t);
-typedef int (*get_metadata_c_local_func)(llama_embedder *,MetadataPair**, size_t*);
-typedef void (*free_metadata_c_local_func)(MetadataPair*, size_t);
-
+    typedef llama_embedder* (*init_embedder_local_func)(const char*, uint32_t);
+    typedef void (*free_embedder_local_func)(llama_embedder*);
+    typedef FloatMatrix (*embed_c_local_func)(llama_embedder*, const char**, size_t, int32_t);
+    typedef int (*get_metadata_c_local_func)(llama_embedder*, MetadataPair**, size_t*);
+    typedef void (*free_metadata_c_local_func)(MetadataPair*, size_t);
 #endif
 
 

--- a/bindings/go/wrapper.cpp
+++ b/bindings/go/wrapper.cpp
@@ -208,14 +208,16 @@ int init_llama_embedder(char * model_path, uint32_t pooling_type ) {
 }
 
 void free_llama_embedder() {
-    if (embedder) {
+    if (embedder != nullptr) {
         free_embedder_f(embedder);
     }
     if (libh != nullptr) {
 #if defined(_WIN32) || defined(_WIN64)
+        fprintf(stderr, "Freeing library under win\n");
         if (!FreeLibrary(libh)){
             fprintf(stderr, "Failed to free library %lu\n", GetLastError());
         }
+        frprintf(stderr, "FREEED library under win\n");
 #else
         if(dlclose(libh) != 0){
             fprintf(stderr, "Failed to close library %s\n", dlerror());

--- a/bindings/go/wrapper.cpp
+++ b/bindings/go/wrapper.cpp
@@ -40,11 +40,24 @@ std::string GetLastErrorAsString() {
 }
 #endif
 
+#if defined(_WIN32) || defined(_WIN64)
+
+typedef llama_embedder * (__cdecl *init_embedder_local_func)(const char *, uint32_t);
+typedef void (__cdecl *free_embedder_local_func)(llama_embedder *);
+typedef FloatMatrix (__cdecl *embed_c_local_func)(llama_embedder *, const char  ** , size_t , int32_t);
+typedef int (__cdecl *get_metadata_c_local_func)(llama_embedder *,MetadataPair**, size_t*);
+typedef void (__cdecl *free_metadata_c_local_func)(MetadataPair*, size_t);
+
+#else
+
 typedef llama_embedder * (*init_embedder_local_func)(const char *, uint32_t);
 typedef void (*free_embedder_local_func)(llama_embedder *);
 typedef FloatMatrix (*embed_c_local_func)(llama_embedder *, const char  ** , size_t , int32_t);
 typedef int (*get_metadata_c_local_func)(llama_embedder *,MetadataPair**, size_t*);
 typedef void (*free_metadata_c_local_func)(MetadataPair*, size_t);
+
+#endif
+
 
 lib_handle libh = nullptr;
 llama_embedder * embedder = nullptr;
@@ -82,13 +95,13 @@ lib_handle load_library(const char * shared_lib_path){
 #endif
 
 #if defined(_WIN32) || defined(_WIN64)
-        init_embedder_f = (init_embedder_local_func) GetProcAddress(libh, "init_embedder");
+        init_embedder_f = reinterpret_cast<init_embedder_local_func>(GetProcAddress(libh, "init_embedder"));
         if (!init_embedder_f) {
             std::string error_message = "Failed to load init_embedder function: " + GetLastErrorAsString();
             throw std::runtime_error(error_message);
         }
 #else
-        init_embedder_f = (init_embedder_local_func) dlsym(libh, "init_embedder");
+        init_embedder_f = reinterpret_cast<init_embedder_local_func>(dlsym(libh, "init_embedder"));
         if (!init_embedder_f) {
             std::string error_message = "Failed to load init_embedder function: " + std::string(dlerror());
             throw std::runtime_error(error_message);
@@ -96,13 +109,13 @@ lib_handle load_library(const char * shared_lib_path){
 #endif
 
 #if defined(_WIN32) || defined(_WIN64)
-        free_embedder_f = (free_embedder_local_func) GetProcAddress(libh, "free_embedder");
+        free_embedder_f = reinterpret_cast<free_embedder_local_func>(GetProcAddress(libh, "free_embedder"));
         if (!free_embedder_f) {
             std::string error_message = "Failed to load free_embedder function: " + GetLastErrorAsString();
             throw std::runtime_error(error_message);
         }
 #else
-        free_embedder_f = (free_embedder_local_func) dlsym(libh, "free_embedder");
+        free_embedder_f = reinterpret_cast<free_embedder_local_func>(dlsym(libh, "free_embedder"));
         if (!free_embedder_f) {
             std::string error_message = "Failed to load free_embedder function: " + std::string(dlerror());
             throw std::runtime_error(error_message);
@@ -110,13 +123,13 @@ lib_handle load_library(const char * shared_lib_path){
 #endif
 
 #if defined(_WIN32) || defined(_WIN64)
-        embed_f = (embed_c_local_func) GetProcAddress(libh, "embed_c");
+        embed_f = reinterpret_cast<embed_c_local_func>(GetProcAddress(libh, "embed_c"));
         if (!embed_f) {
             std::string error_message = "Failed to load embed function: " + GetLastErrorAsString();
             throw std::runtime_error(error_message);
         }
 #else
-        embed_f = (embed_c_local_func) dlsym(libh, "embed_c");
+        embed_f = reinterpret_cast<embed_c_local_func>(dlsym(libh, "embed_c"));
         if (!embed_f) {
             std::string error_message = "Failed to load embed function: " + std::string(dlerror());
             throw std::runtime_error(error_message);
@@ -124,13 +137,13 @@ lib_handle load_library(const char * shared_lib_path){
 #endif
 
 #if defined(_WIN32) || defined(_WIN64)
-        get_metadata_f = (get_metadata_c_local_func) GetProcAddress(libh, "get_metadata_c");
+        get_metadata_f = reinterpret_cast<get_metadata_c_local_func>(GetProcAddress(libh, "get_metadata_c"));
         if (!get_metadata_f) {
             std::string error_message = "Failed to load get_metadata function: " + GetLastErrorAsString();
             throw std::runtime_error(error_message);
         }
 #else
-        get_metadata_f = (get_metadata_c_local_func) dlsym(libh, "get_metadata_c");
+        get_metadata_f = reinterpret_cast<get_metadata_c_local_func>( dlsym(libh, "get_metadata_c"));
         if (!get_metadata_f) {
             std::string error_message = "Failed to load get_metadata function: " + std::string(dlerror());
             throw std::runtime_error(error_message);
@@ -139,13 +152,13 @@ lib_handle load_library(const char * shared_lib_path){
 
 
 #if defined(_WIN32) || defined(_WIN64)
-        free_metadata_f = (free_metadata_c_local_func) GetProcAddress(libh, "free_metadata_c");
+        free_metadata_f = reinterpret_cast<free_metadata_c_local_func>(GetProcAddress(libh, "free_metadata_c"));
         if (!free_metadata_f) {
             std::string error_message = "Failed to load free_metadata function: " + GetLastErrorAsString();
             throw std::runtime_error(error_message);
         }
 #else
-        free_metadata_f = (free_metadata_c_local_func) dlsym(libh, "free_metadata_c");
+        free_metadata_f = reinterpret_cast<free_metadata_c_local_func>(dlsym(libh, "free_metadata_c"));
         if (!free_metadata_f) {
             std::string error_message = "Failed to load free_metadata function: " + std::string(dlerror());
             throw std::runtime_error(error_message);

--- a/bindings/go/wrapper.h
+++ b/bindings/go/wrapper.h
@@ -4,11 +4,11 @@
 #ifndef WRAPPER_H
 #define WRAPPER_H
 #if defined(_WIN32) || defined(_WIN64)
-#define EXPORT_SYMBOL __declspec(dllexport)
+#define EXPORT_GO_WRAPPER __declspec(dllexport)
 #include <windows.h>
     typedef HMODULE lib_handle;
 #else
-#define EXPORT_SYMBOL __attribute__((visibility("default")))
+#define EXPORT_GO_WRAPPER __attribute__((visibility("default")))
 #include <dlfcn.h>
     typedef void* lib_handle;
 #endif
@@ -28,16 +28,16 @@ typedef struct {
     size_t cols;
 } FloatMatrixW;
 
-EXPORT_SYMBOL lib_handle load_library(const char *shared_lib_path);
-EXPORT_SYMBOL int init_llama_embedder(char *model_path, uint32_t pooling_type);
-EXPORT_SYMBOL void free_llama_embedder();
-EXPORT_SYMBOL FloatMatrixW llama_embedder_embed(const char **texts, size_t text_count, int32_t norm);
-EXPORT_SYMBOL void free_float_matrixw(FloatMatrixW * fm);
+EXPORT_GO_WRAPPER lib_handle load_library(const char *shared_lib_path);
+EXPORT_GO_WRAPPER int init_llama_embedder(char *model_path, uint32_t pooling_type);
+EXPORT_GO_WRAPPER void free_llama_embedder();
+EXPORT_GO_WRAPPER FloatMatrixW llama_embedder_embed(const char **texts, size_t text_count, int32_t norm);
+EXPORT_GO_WRAPPER void free_float_matrixw(FloatMatrixW * fm);
 
-EXPORT_SYMBOL const char* get_last_error();
+EXPORT_GO_WRAPPER const char* get_last_error();
 
-EXPORT_SYMBOL char ** llama_embedder_get_metadata(size_t* size);
-EXPORT_SYMBOL void free_metadata(char** metadata_array, size_t size);
+EXPORT_GO_WRAPPER char ** llama_embedder_get_metadata(size_t* size);
+EXPORT_GO_WRAPPER void free_metadata(char** metadata_array, size_t size);
 
 #ifdef __cplusplus
 }

